src/spamfilter.c

```diff
@@ -488,16 +488,33 @@ void stripcolors(char new[512], char *org)
 {
     int len = 0;
 
-    for(; (*org && len<512); org++)
+    for(; (*org && len < 512); org++)
     {
-        if(*org=='\003')
+        if(*org == '\003')
         {
+            // Color codes: 1-2 digits, then optionally followed by a comma and an additional 1-2 digits
             org++;
-            while(IsDigit(*org) || *org==',')
+            if(*org && IsDigit(*org))
```

Comment:
There is no point in checking *org if you're doing IsDigit() right after it... same to the other places, you would only want to check it if you're doing things like *(org + 1) (which would be 0x1 if org is NULL) without checking *org first...

Kobi.